modules: memfault-firmware-sdk: migrate coredump from PM to devicetree (#28820)
nrfconnect/sdk-nrf: snippets/memfault-coredump/README.rst

.. _snippet-memfault-coredump:

Memfault coredump snippet (memfault-coredump)
#############################################

.. code-block:: console

   west build -S memfault-coredump [...]

Overview
********

This snippet enables flash-backed Memfault coredump storage without requiring Partition Manager.

> It sets :kconfig:option:`CONFIG_MEMFAULT_NCS_INTERNAL_FLASH_BACKED_COREDUMP` and applies a per-board devicetree overlay that defines a fixed partition labeled ``memfault_coredump_partition`` within internal flash.

Reviewer: ```suggestion
It sets the :kconfig:option:`CONFIG_MEMFAULT_NCS_INTERNAL_FLASH_BACKED_COREDUMP` Kconfig option and applies a board-specific devicetree overlay that defines a fixed partition labeled ``memfault_coredump_partition`` within internal flash.
```

Supported boards
****************

* ``nrf52840dk/nrf52840``
* ``nrf5340dk/nrf5340/cpuapp`` and ``nrf5340dk/nrf5340/cpuapp/ns``
* ``nrf9160dk/nrf9160/ns``
* ``nrf9151dk/nrf9151/ns``
* ``nrf9161dk/nrf9161/ns``
* ``thingy91/nrf9160/ns``
* ``thingy91x/nrf9151/ns``

For other boards, add a ``fixed-partition`` node labeled ``memfault_coredump_partition`` under the internal flash controller in your overlay file, ensuring the size is aligned to a multiple of the flash erase block (typically 4 kB).

Notes
*****

> When using sysbuild, also set ``SB_CONFIG_PARTITION_MANAGER=n`` (for example, using a ``Kconfig.sysbuild`` fragment in your application) so that Partition Manager is disabled at the sysbuild level as well.

Reviewer: ```suggestion
When using sysbuild, also set the :kconfig:option:`SB_CONFIG_PARTITION_MANAGER` Kconfig option to ``n`` (for example, using a ``Kconfig.sysbuild`` fragment in your application) so that Partition Manager is disabled at the sysbuild level as well.
```

> When using sysbuild, also set the :kconfig:option:`SB_CONFIG_PARTITION_MANAGER` Kconfig option to ``n`` (for example, using a ``Kconfig.sysbuild`` fragment in your application) so that Partition Manager is disabled at the sysbuild level as well.

Reviewer 2: sysbuild is always enabled by default, who would be building without it?

Author: good point, removed this language :+1:

> The nRF91 ``ns`` overlays redefine the full flash and modem-shared SRAM layout because disabling Partition Manager removes the TF-M and modem partition generation that |NCS| would otherwise generate.

Reviewer: ```suggestion
The nRF91 ``ns`` overlays redefine the full flash and modem-shared SRAM layout because disabling Partition Manager removes the TF-M and modem partition generation that the |NCS| would otherwise generate.
```

.. _snippet-memfault-coredump:

Memfault coredump snippet (memfault-coredump)
#############################################

.. code-block:: console

   west build -S memfault-coredump [...]

Overview
********

This snippet enables flash-backed Memfault coredump storage without requiring
Partition Manager. It sets :kconfig:option:`CONFIG_MEMFAULT_NCS_INTERNAL_FLASH_BACKED_COREDUMP`
and applies a per-board devicetree overlay that defines a fixed partition

> labelled ``memfault_coredump_partition`` within internal flash.

Reviewer 3: ```suggestion
This snippet enables flash-backed Memfault coredump storage without requiring Partition Manager. 
It sets :kconfig:option:`CONFIG_MEMFAULT_NCS_INTERNAL_FLASH_BACKED_COREDUMP` and applies a per-board devicetree overlay that defines a fixed partition labeled ``memfault_coredump_partition`` within internal flash.
```

> labelled ``memfault_coredump_partition`` at the end of internal flash.

Reviewer 4: Is there a reason for it to be specifically at the end of internal flash? would a different place in internal flash or even external flash also work?

Author: good question- definitely doesn't matter where in internal flash it's placed, i'll update this note 👍 .

i think external flash deserves a harder look (on my to-do list), because there might be issues accessing from exception context. i'll report back when i've done some digging 🫡

Author: fixed the description here to say internal flash for now (removed "at the end of")

Supported boards
****************

* ``nrf52840dk/nrf52840``
* ``nrf5340dk/nrf5340/cpuapp`` and ``nrf5340dk/nrf5340/cpuapp/ns``
* ``nrf9160dk/nrf9160/ns``
* ``nrf9151dk/nrf9151/ns``
* ``nrf9161dk/nrf9161/ns``
* ``thingy91/nrf9160/ns``
* ``thingy91x/nrf9151/ns``

For other boards, add a ``fixed-partition`` node labelled
``memfault_coredump_partition`` under the internal flash controller in your

> own overlay, sized to a multiple of the flash erase block (typically 4 kB).

Reviewer 3: ```suggestion
For other boards, add a ``fixed-partition`` node labeled ``memfault_coredump_partition`` under the internal flash controller in your overlay file, ensuring the size is aligned to a multiple of the flash erase block (typically 4 kB).
```

Notes
*****

When using sysbuild, also set ``SB_CONFIG_PARTITION_MANAGER=n`` (for example
via a ``Kconfig.sysbuild`` fragment in your application) so that Partition

> Manager is disabled at the sysbuild level as well.

Reviewer 3: ```suggestion
When using sysbuild, also set ``SB_CONFIG_PARTITION_MANAGER=n`` (for example, using a ``Kconfig.sysbuild`` fragment in your application) so that Partition Manager is disabled at the sysbuild level as well.
```

The nRF91 NS overlays redefine the full flash and modem-shared SRAM layout
because disabling Partition Manager removes the TF-M and modem partition
generation that NCS would otherwise inject. The layouts mirror those used by

> the ``samples/cellular/at_client`` sample.

Reviewer 3: ```suggestion
The nRF91 ``ns`` overlays redefine the full flash and modem-shared SRAM layout because disabling Partition Manager removes the TF-M and modem partition generation that |NCS| would otherwise generate. 
The layouts mirror those used by the :ref:`at_client_sample` sample.
```
